lib/src/clojure_lsp/kondo.clj

```diff
@@ -74,9 +74,19 @@
       (update :findings merge findings)
       (assoc :kondo-config config)))
 
-(defn entry->normalized-entries [{:keys [bucket arglist-strs] :as element}]
-  (cond
-    (identical? :var-definitions bucket)
+;;;; Normalization
+
+;; The analysis we get back from clj-kondo is indexed by bucket
+;; {(bucket [element+])*}
+;; We re-index by filename then bucket
+;; {(filename {(bucket [element+])+})*}
+;; We also normalize elements somewhat, changing their keys, and even spliting
+;; some into two elements. As such, the buckets we get from clj-kondo aren't
+;; exactly the same as the buckets we have in our db.
```

Comment:
This may be interesting for @borkdude, I wonder if clj-kondo could do that for us already in a way we don't need breaking changes, like a new flag or something, I can see this being very convenient for clj-kondo analysis clients for performance reasons